app/services/bulk_create_test_applications.rb

```diff
@@ -0,0 +1,69 @@
+class BulkCreateTestApplications
+  attr_reader :template_application_form, :candidate, :application_choice
+
+  def initialize(template_application_form)
+    @template_application_form = template_application_form
+    @candidate = template_application_form.candidate
+    @application_choice = template_application_form.application_choices.first
+  end
+
+  def call
+    random_id = SecureRandom.hex(10)
```

Comment:
Should we look up if the ID is already in use and retry if so?

Reply:
Rather than add the additional query, a couple of alternatives might be:

- Let it crash out on the email unique index on the rare occasion that there's a conflict, and just restart is manually with however many records you have left to create
- Switch to using UUIDs. Significantly uglier in the UI but extremely low chance of conflict.

Reply:
Hmm. Instead of an additional query to check if it's available, could we wrap the `ActiveRecord.execute` with `begin/rescue` and if we rescue from the `RecordAlreadyExistsWhatchacallit` then we re-assign `random_id = SecureRandom.hex(10)` again, upwards of 3 tries or so?